src/libs/auth.ts

```diff
@@ -109,59 +149,146 @@ const getSigninArgs = (includeBillingScope) => {
+      ...state,
+      authTokenMetadata: {
+        ...authStore.get().authTokenMetadata,
+        createdAt: authTokenCreatedAt,
+        expiresAt: authTokenExpiresAt,
+        id: authTokenId,
+        totalAuthTokensUsedThisSession: authStore.get().authTokenMetadata.totalAuthTokensUsedThisSession + 1,
+      },
+    }));
+    Ajax().Metrics.captureEvent(Events.user.authTokenLoad.success, {
+      authProvider: user.profile.idp,
+      ...labelOldAuthToken(oldAuthTokenMetadata),
+      authTokenCreatedAt: labelTimestampMetric(authTokenCreatedAt),
+      authTokenExpiresAt: labelTimestampMetric(authTokenExpiresAt),
+      authTokenId,
+      refreshTokenId: authStore.get().refreshTokenMetadata.id,
+      refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+      refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+      jwtExpiresAt: labelTimestampMetric(jwtExpiresAt),
+    });
+  } else if (reloadedAuthTokenState === null) {
+    Ajax().Metrics.captureEvent(Events.user.authTokenLoad.expired, {
+      ...labelOldAuthToken(oldAuthTokenMetadata),
+      refreshTokenId: authStore.get().refreshTokenMetadata.id,
+      refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+      refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+    });
+  } else {
+    Ajax().Metrics.captureEvent(Events.user.authTokenLoad.error, {
+      // we could potentially log the reason, but I don't know if that data is safe to log
+      ...labelOldAuthToken(oldAuthTokenMetadata),
+      refreshTokenId: authStore.get().refreshTokenMetadata.id,
+      refreshTokenCreatedAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.createdAt),
+      refreshTokenExpiresAt: labelTimestampMetric(authStore.get().refreshTokenMetadata.expiresAt),
+    });
+  }
+  return reloadedAuthTokenState;
+};
+const labelOldAuthToken = (oldAuthTokenMetadata) => {
+  return {
+    oldAuthTokenCreatedAt: labelTimestampMetric(oldAuthTokenMetadata.createdAt),
+    oldAuthTokenExpiresAt: labelTimestampMetric(oldAuthTokenMetadata.expiresAt),
+    oldAuthTokenId: oldAuthTokenMetadata.id === undefined ? 'undefined' : oldAuthTokenMetadata.id,
+    oldAuthTokenLifespan:
+      oldAuthTokenMetadata.id === undefined ? 'undefined' : Date.now() - oldAuthTokenMetadata.createdAt / 1000.0,
+  };
+};
+const labelTimestampMetric = (timestamp: number): string => {
+  return timestamp < 0 ? 'timestamp unset' : Utils.formatTimestampInSeconds(timestamp);
 };
 
-export const reloadAuthToken = (includeBillingScope = false) => {
+const tryLoadAuthTokenSilent = (includeBillingScope = false): Promise<User | null | boolean> => {
   const args = getSigninArgs(includeBillingScope);
-  const tokenMetadata = authStore.get().authTokenMetadata;
-  let authReloadSuccessful = true;
+  // if this returns a null -> refresh token has expired
```

Comment:
Suggest putting this in JSDoc for the function, so that it's more readily available when writing code that uses this function.